docs/storybook/src/VirtualCursor.ts

```diff
@@ -0,0 +1,154 @@
+      if (!down) cursor.classList.remove("down");
+    };
+    setCursorPosition(this.top, this.left);
+    setCursorDown(false);
+
+    const getNewPosition = (e: MouseEvent) => {
+      const left = e.pageX - cursorSize / 2;
+      const top = e.pageY - cursorSize / 2;
+      return { left, top };
+    };
+
+    const handleMouseMove = (e: MouseEvent) => {
+      const { left, top } = getNewPosition(e);
+      setCursorPosition(top, left);
+    };
+    const handleMouseDown = (e: MouseEvent) => {
+      const { left, top } = getNewPosition(e);
+      setCursorPosition(top, left);
+      setCursorDown(true);
+    };
+    const handleMouseUp = (e: MouseEvent) => {
+      const { left, top } = getNewPosition(e);
+      setCursorPosition(top, left);
+      setCursorDown(false);
+    };
+    document.addEventListener("mousemove", handleMouseMove);
+    document.addEventListener("mousedown", handleMouseDown);
+    document.addEventListener("mouseup", handleMouseUp);
+
+    const stopPropagation = (e: MouseEvent) => {
+      e.stopPropagation();
+    };
+    overlay.addEventListener("mousemove", stopPropagation);
+    overlay.addEventListener("mousedown", stopPropagation);
+    overlay.addEventListener("mouseup", stopPropagation);
+
+    this._removeListeners.push(
+      () => document.removeEventListener("mousemove", handleMouseMove),
+      () => document.removeEventListener("mousedown", handleMouseDown),
+      () => document.removeEventListener("mouseup", handleMouseUp),
+      () => overlay.removeEventListener("mousemove", stopPropagation),
+      () => overlay.removeEventListener("mousedown", stopPropagation),
+      () => overlay.removeEventListener("mouseup", stopPropagation)
+    );
+
+    shadow.appendChild(style);
+    shadow.appendChild(overlay);
+    overlay.appendChild(cursor);
+  }
+
+  public disconnectedCallback() {
+    for (const removeListener of this._removeListeners) {
+      removeListener();
+    }
+  }
+}
+
+export function createCursorEvents(
+  initialPosition: XAndY,
+  onMove: (to: XAndY) => void,
+  options?: {
+    duration?: number;
+  }
+) {
+  let currentPosition = initialPosition;
+  const duration = options?.duration ?? 250;
+  return {
+    move: async (to: XAndY) => {
+      let start = Date.now();
+      const line = LineSegment3d.create(
+        new Point3d(currentPosition.x, currentPosition.y, 0),
+        new Point3d(to.x, to.y, 0)
+      );
+      return new Promise<void>((resolve) => {
+        const move = () => {
+          const now = Date.now();
+          const elapsed = now - start;
+          const fraction = elapsed / duration;
+          if (fraction < 1) {
+            const position = line.fractionToPoint(fraction);
+            onMove({ x: position.x, y: position.y });
+            requestAnimationFrame(move);
+            return;
+          }
+
+          onMove(to);
+          currentPosition = to;
+          resolve();
+        };
+        requestAnimationFrame(move);
+      });
+    },
```

Comment:
Could we replace that by simply putting a `transition: transform 1s ease-in-out` ? (Or maybe have the 1s a parameter of the move)

Reply:
I've tried that idea initially, however imagine dragging a